draft-packages/modal/KaizenDraft/Modal/Presets/ConfirmationModal.tsx

```diff
@@ -58,48 +58,53 @@ const ConfirmationModal = ({
   dismissLabel = "Cancel",
   automationId,
   children,
-}: Props) => (
-  <GenericModal
-    isOpen={isOpen}
-    onEscapeKeyup={onDismiss}
-    onOutsideModalClick={onDismiss}
-    automationId={automationId}
-  >
-    <div className={styles.modal}>
-      <ModalHeader unpadded reversed onDismiss={onDismiss}>
-        <div
-          className={classnames(styles.header, {
-            [styles.cautionaryHeader]: type === "cautionary",
-            [styles.informativeHeader]: type === "informative",
-            [styles.negativeHeader]: type === "negative",
-            [styles.positiveHeader]: type === "positive",
-          })}
-        >
-          <div className={styles.iconContainer}>
-            <div className={styles.spotIcon}>{getIcon(type)}</div>
+}: Props) => {
+  const footerActions: Array<{ label: string; action: () => void }> = []
+  if (onConfirm) {
+    footerActions.push({ label: confirmLabel, action: onConfirm })
+  }
+  footerActions.push({ label: dismissLabel, action: onDismiss })
```

Comment:
The only update in this block of code is this `footerActions` constant, which then gets passed down to the `ModalFooter` component